### PR TITLE
Cleanup DG order zero, use `index_range`

### PR DESCRIPTION
**Pull request by dschwen**
Refs #21316

### CODE REVIEW

File: framework/include/utils/MooseTypes.h [hunk 1]
```diff
@@ -963,3 +964,10 @@ extern const TagName PREVIOUS_NL_SOLUTION_TAG;
   {                                                                                                \
     __VA_ARGS__                                                                                    \
   }
+
+template <typename T>
+auto
+index_range(const T & sizable)
+{
+  return IntRange<decltype(sizable.size())>(0, sizable.size());
+}
```
Comment (dschwen) [inline framework/include/utils/MooseTypes.h:973]: See also libmesh/libmesh#3377

**Review by loganharbour — APPROVED** [review]

**Comment by dschwen** [comment]
Ill go ahead and merge this. Once https://github.com/libMesh/libmesh/pull/3378 goes in and MOOSE gets a libmesh submodule update we can remove the template copy in MOOSE.

This:

image

is an approval to merge, right?

**Comment by dschwen** [comment]
Oh, never mind. Now you effectively disabled merges. Somebody is updating Okami's Bison sub,odule?

**Comment by roystgnr** [comment]
> we can remove the template copy in MOOSE.

We need to temporarily rename the template copy in MOOSE, unless we want to revert https://github.com/libMesh/libmesh/pull/3378 - currently MOOSE won't compile against libMesh master, including MOOSE CI.  We can rename the use cases back when we do the next submodule merge.

**Pull request merged**